### PR TITLE
DMP-4014 - Remove relation from hearing_event_ae table when event is …

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
…made is_current false
https://tools.hmcts.net/jira/browse/DMP-4014

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/EventStub.java [hunk 1]
```diff
@@ -162,4 +165,17 @@ public boolean isOnlyOneOfTheEventIdSetToCurrent(List<EventEntity> eventEntityLi
 
         return currentFnd;
     }
+
+    public boolean isOnlyOneOfTheEventIdGotHearings(List<EventEntity> eventEntityList) {
```
Comment (karen-hedges) [inline src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/EventStub.java:169]: Should be has not is?

File: src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java [hunk 2]
```diff
@@ -31,6 +31,14 @@ public interface HearingRepository extends JpaRepository<HearingEntity, Integer>
     )
     List<HearingEntity> findByCaseIds(List<Integer> caseIds);
 
+    @Query("""
+        SELECT h.id FROM HearingEntity h
+        JOIN h.eventList event
+        WHERE event.id = :eventId
+        """
+    )
+    List<Integer> findHearingIdsByEventId(Integer eventId);
```
Comment (karen-hedges) [inline src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java:40]: Are you going to create a test for this query like EventRepositoryTest?

File: src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java [hunk 3]
```diff
@@ -97,7 +97,7 @@ SELECT e.event_id, he.eve_id, string_agg(he.hea_id::varchar, ',' order by he.hea
         having count(e2.event_id) > 1
         limit :limit
         """, nativeQuery = true)
-    List<Integer> getCurrentEventIdsToBeProcessed(long limit);
+    List<Integer> findCurrentEventIdsWithDuplicates(long limit);
```
Comment (karen-hedges) [inline src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java:100]: Is it including duplicates or just those events that are duplicates?

File: src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/EventStub.java (hunk 1 above)
Comment (cmagowan-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/EventStub.java]: done

File: src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java (hunk 2 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java:40]: done

File: src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java:100]: one eventId can have multiple duplicate events, so this is returning the eventIds with more than 1 event, not the ids of the events. eventId is not to be confused with the id of the event table (eventEntity.id), this is really badly named and does cause confusion.

**Review by karen-hedges — APPROVED** [review]

**Review by davet1985 — APPROVED** [review]

**Pull request merged**